instrumentation/jdbc/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcUtils.java

```diff
@@ -69,25 +74,35 @@ public static Connection connectionFromStatement(Statement statement) {
     return connection;
   }
 
-  /** Returns null if the sql could not be normalized for any reason. */
-  public static String normalizeSql(String sql) {
+  public static SqlStatementInfo sanitizeAndExtractInfo(String sql) {
+    try {
+      return sqlToStatementInfoCache.get(
+          sql,
+          () -> {
+            log.debug("SQL statement cache miss");
+            try {
+              return SqlStatementInfoExtractor.extract(sanitizeSql(sql));
+            } catch (ParseException e) {
+              log.debug("Could not extract sql info", e);
+              return new SqlStatementInfo(null, null, null);
```

Comment:
Can we use a single pre-made instance for this?

Reply:
I don't think that's necessary - this is a "should never happen" scenario, as the lexer actually matches all possible input.